gap/Utils.gi

```diff
@@ -136,6 +136,67 @@ function(type, alpha, q)
     fi;
 end);
 
+# Find gamma in GF(q) such that x ^ 2 + x + gamma is irreducible for q a power
+# of two.
+BindGlobal("FindGamma",
+function(q)
+    local F, R, x, gamma, polynomial;
+
+    if not IsEvenInt(q) then
+        ErrorNoReturn("<q> must be even");
+    elif not IsPrimePowerInt(q) then
+        ErrorNoReturn("<q> must be a power of 2");
+    fi;
+
+    F := GF(q);
+    R := PolynomialRing(F, ["x"]);
+    x := Indeterminate(F, "x");
+
+    for gamma in F do
+        polynomial := x ^ 2 + x + gamma;
+        if IsIrreducibleRingElement(R, polynomial) then
+            return gamma;
+        fi;
+    od;
```

Comment:
That's a brute force search which seems inefficient. Can't  you adapt the idea from `SolveQuadraticEquation`: the map `t -> t^2 + t` is linear in char 2. You are looking for value `gamma` that is *not* in its image. So like this:

```gap
FindGamma2 := function(q)
    local F, B, M, i;

    F := GF(q);
    B := CanonicalBasis(F);
    M := List(B, b -> Coefficients(B, b + b ^ 2));
    ConvertToMatrixRep(M);
    M := RREF(M);
    i := First([1..Length(M)], i -> IsZero(M[i,i]));
    return B[i];
end;
```
For q=2^17 the original `FindGamma` runs a long time for me, because there is no conway polynomial and no good polynomial factoring is available in that field in GAP; while `FindGamma2` is instant.

Reply:
Okay, this makes sense: It's not the actual brute force search that's inefficient (because half of all possible `gamma` make `x ^ 2 + x + gamma` irreducible by a simple linear algebra argument), but the test for irreducibility involved in the process! ;)